lib/services/okta_test.go

```diff
@@ -0,0 +1,209 @@
+						},
+					},
+					AddLabels: map[string]string{
+						"label1": "value1",
+					},
+				},
+				{
+					Match: []*types.OktaImportRuleMatchV1{
+						{
+							GroupIDs: []string{"yes"},
+						},
+					},
+					AddLabels: map[string]string{
+						"label1": "value1",
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+	data, err := MarshalOktaImportRule(expected)
+	require.NoError(t, err)
+	actual, err := UnmarshalOktaImportRule(data)
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+// TestOktaAssignmentUnmarshal verifies an Okta assignment resource can be unmarshaled.
+func TestOktaAssignmentUnmarshal(t *testing.T) {
+	expected, err := types.NewOktaAssignment(
+		types.Metadata{
+			Name: "test-assignment",
+		},
+		types.OktaAssignmentSpecV1{
+			User: "test-user@test.user",
+			Actions: []*types.OktaAssignmentActionV1{
+				{
+					Status: types.OktaAssignmentActionV1_PENDING,
+					Target: &types.OktaAssignmentActionTargetV1{
+						Type: types.OktaAssignmentActionTargetV1_APPLICATION,
+						Id:   "123456",
+					},
+				},
+				{
+					Status: types.OktaAssignmentActionV1_SUCCESSFUL,
+					Target: &types.OktaAssignmentActionTargetV1{
+						Type: types.OktaAssignmentActionTargetV1_GROUP,
+						Id:   "234567",
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+	data, err := utils.ToJSON([]byte(oktaAssignmentYAML))
+	require.NoError(t, err)
+	actual, err := UnmarshalOktaAssignment(data)
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+// TestOktaAssignmentMarshal verifies a marshaled Okta assignment resource can be unmarshaled back.
+func TestOktaAssignmentMarshal(t *testing.T) {
+	expected, err := types.NewOktaAssignment(
+		types.Metadata{
+			Name: "test-assignment",
+		},
+		types.OktaAssignmentSpecV1{
+			User: "test-user@test.user",
+			Actions: []*types.OktaAssignmentActionV1{
+				{
+					Status: types.OktaAssignmentActionV1_PENDING,
+					Target: &types.OktaAssignmentActionTargetV1{
+						Type: types.OktaAssignmentActionTargetV1_APPLICATION,
+						Id:   "123456",
+					},
+				},
+				{
+					Status: types.OktaAssignmentActionV1_SUCCESSFUL,
+					Target: &types.OktaAssignmentActionTargetV1{
+						Type: types.OktaAssignmentActionTargetV1_GROUP,
+						Id:   "234567",
+					},
+				},
+			},
+		},
+	)
+	require.NoError(t, err)
+	data, err := MarshalOktaAssignment(expected)
+	require.NoError(t, err)
+	actual, err := UnmarshalOktaAssignment(data)
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+var (
+	oktaImportRuleYAML = `---
+kind: okta_import_rule
+version: v1
+metadata:
+  name: test-rule
+spec:
+  mappings:
+  - match:
+    - app_ids: ["yes"]
+    add_labels:
+      label1: value1
+  - match:
+    - group_ids: ["yes"]
+    add_labels:
+      label1: value1
+`
+
+	oktaAssignmentYAML = `---
+kind: okta_assignment
+version: v1
+metadata:
+  name: test-assignment
+spec:
+  user: test-user@test.user
+  actions:
+  - status: 1
```

Comment:
sorry, i forgot ask about it in main PR, but is it common to use enum as value in yaml configs? I always preferred using string value as it's more readable.

Reply:
My guess is that it's uncommon, but these are expected to be mostly used behind the scenes. That being said, I'm glad you asked, because I was wondering the same thing. I'll track this down and submit a follow up if I can.